libherokubuildpack/src/download.rs

```diff
@@ -21,7 +21,12 @@ pub enum DownloadError {
     CannotParseInteger(ParseIntError),
 
     #[error("Expected `{expected}` bytes received `{received}`")]
-    UnexpectedBytes { expected: u64, received: u64 },
+    UnexpectedBytes {
+        expected: u64,
+        received: u64,
+        path_deleted: Result<(), std::io::Error>,
```

Comment:
I got a little clever with this representation. I'm not sure if this is a net good or bad idea (to return a Result here).

My logic for this is: It seems unlikely that the delete would fail if we were able to create and write to the file, however, it's still possible so we need to represent it. It seems wrong to return an `DownloadError::IoError` as that explicitly says it's for io errors during download (as opposed to cleanup). We could add another variant just for this case, but it would end up having almost the same information. 

The primary error here is the bytes not written, whether or not the disk could be cleaned is a secondary concern.

I'm open to alternative suggestions.